Regression liquibase 4.25.0: Invalid query generated by Liquibase#update

### Search first

- [X] I searched and no similar issues were found

### Description

After upgrade to 4.25, calling Liquibase#update produces a java.sql.SQLSyntaxErrorException for a jdbc query with incorrect parameters mappings:

```
Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.UnexpectedLiquibaseException: liquibase.exception.DatabaseException: java.sql.SQLSyntaxErrorException: (conn=279) Could not set parameter at position 2 (values was 'DATABASECHANGELOG')
Query - conn:279(M)  - "SELECT TABLE_SCHEMA, TABLE_NAME, COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = ?"
        at deployment.nitro-ear.ear//liquibase.command.CommandScope.execute(CommandScope.java:237)
        at deployment.nitro-ear.ear//liquibase.Liquibase.lambda$update$0(Liquibase.java:242)
        at deployment.nitro-ear.ear//liquibase.Scope.lambda$child$0(Scope.java:184)
        at deployment.nitro-ear.ear//liquibase.Scope.child(Scope.java:193)
        at deployment.nitro-ear.ear//liquibase.Scope.child(Scope.java:183)
        at deployment.nitro-ear.ear//liquibase.Scope.child(Scope.java:162)
        at deployment.nitro-ear.ear//liquibase.Liquibase.runInScope(Liquibase.java:1403)
        at deployment.nitro-ear.ear//liquibase.Liquibase.update(Liquibase.java:231)
        at deployment.nitro-ear.ear//liquibase.Liquibase.update(Liquibase.java:209)
        at deployment.nitro-ear.ear//liquibase.Liquibase.update(Liquibase.java:191)
        at deployment.nitro-ear.ear//liquibase.Liquibase.update(Liquibase.java:176)
        at deployment.nitro-ear.ear.nitro-core-services.jar//be.fiscalteam.nitro.util.provision.LiquibaseService.runChangelog(LiquibaseService.java:41)
```

Notice that only 1 parameter is provided in the query, and liquibase attempts to map parameter #2.

Rolling back to 4.24 works around the issue.

The custom code calling liquibase uses an injected `javax.sql.DataSource` and instantiate Liquibase the following way:

 ```
        try (Connection connection = dataSource.getConnection()) {
            JdbcConnection jdbcConnection = new JdbcConnection(connection);
            DatabaseFactory databaseFactory = DatabaseFactory.getInstance();
            Database database = databaseFactory.findCorrectDatabaseImplementation(jdbcConnection);
            ClassLoader classLoader = getClass().getClassLoader();
            ResourceAccessor classLoaderResourceAccessor = new ClassLoaderResourceAccessor(classLoader);
            ResourceAccessor fileSystemResourceAccessor = new FileSystemResourceAccessor();
            CompositeResourceAccessor resourceAccessor = new CompositeResourceAccessor(classLoaderResourceAccessor, fileSystemResourceAccessor);

            Liquibase liquibase = new Liquibase(changeLogFile, resourceAccessor, database);
            liquibase.update((String) null);
        }
```

The datasource is mapped to a mariadb 10 database.

### Steps To Reproduce

- Initialize a deployment using a mariadb jdbc datasource, and initialize liquibase like in the snippet above

### Expected/Desired Behavior

Liquibase produces valid sql/jsql requests

### Liquibase Version

4.25

### Database Vendor & Version

mariadb 10

### Liquibase Integration

liquibase.Liquibase#update

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

linnux, jakarta 10, wildfly

### Additional Context

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

--- a/liquibase-standard/src/main/java/liquibase/snapshot/JdbcDatabaseSnapshot.java
+++ b/liquibase-standard/src/main/java/liquibase/snapshot/JdbcDatabaseSnapshot.java
@@ -516,11 +516,13 @@ public class JdbcDatabaseSnapshot extends DatabaseSnapshot {
                 //
                 StringBuilder selectStatement = new StringBuilder(
                     "SELECT TABLE_SCHEMA, TABLE_NAME, COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = ?");
+                if(tableName != null) {
+                    selectStatement.append(" AND TABLE_NAME = ?");
+                }
                 Connection underlyingConnection = ((JdbcConnection) database.getConnection()).getUnderlyingConnection();
                 PreparedStatement statement = underlyingConnection.prepareStatement(selectStatement.toString());
                 statement.setString(1, schemaName);
                 if (tableName != null) {
-                    selectStatement.append(" AND TABLE_NAME = ?");
                     statement.setString(2, tableName);
                 }
                 try {

